### ISSUE TITLE
Template class completely broken

### ISSUE THREAD

**Issue by hofrob** [yiisoft/yii2-twig#113]
Please anyone who is working on this take a look at the class `yii\twig\Template`

```php
$object instanceof ElementInterface
```

ElementInterface is not imported. Nor does it exist in any of the dependencies.

```php
self::_includeElementInTemplateCaches($object);
```

This method does not exist.

```php
$object instanceof Object
```

Since PHP 7.2 `object` is a reserved word. The Yii2 class `Object` is not imported anyways.

```php
parent::displayWithErrorHandling($context, $blocks);
```

This class does not extend another class. Calling `parent::something()` *always* fails.

```php
$this->getDebugInfo()
```

This method does not exist.

Honestly we are a little bit puzzled how anyone can actually use this extension.

**Comment by machour** [comment]
I can confirm that the 2.2.x branch looks really odd..

I don't see that much oddness in 2.1.1, could you try downgrading to that version and see what's what?
https://github.com/yiisoft/yii2-twig/tree/2.1.1

**Comment by koxu1996** [comment]
@horfob @machour I use this extension in my personal project, so I can confirm it's fully working. As I [mentioned](https://github.com/yiisoft/yii2-twig/pull/84#issuecomment-318835784) the code was copied from _craftcms_ and adjusted to pass all tests. It seems [these three lines](https://github.com/yiisoft/yii2-twig/blob/806ad21b4edbf3542b532818c74e353d16dbcca6/Template.php#L35-L37) are irrelevant:
```
if ($object instanceof ElementInterface) {
    self::_includeElementInTemplateCaches($object);
}
```
... but _weird_ code inside is not executed - `instanceof` returns `false`. It's really interesting that you can provide any class name, even non-existing one and _PHP_ won't leave you error nor warning.

To sum up **extension is working fine**, but contain few unnecessary lines which doesn't matter. I'll prepare PR to get rid of them.

@hofrob nice catch btw.

**Comment by hofrob** [comment]
Thanks for the quick replys! Will try 2.1.1 and report back. At first glance this does look better.

The version we used had the erroraenous comparison `instanceof Object` (which causes getters to be executed twice) and I was about to do a PR until I saw `yii\twig\Template` in the master branch.

If I have something worth contributing, I will do that.
